### PR TITLE
fix(Call): Fix flash talk indicator

### LINKED ISSUE #821: bug(calling): Voice indicator just blinks
**Issue:**
Seems for some people this works fine but for most it doesnt. It just blinks rapidly when the user is talking. Sometimes for the user speaking no indicator will appear at all 

*Note that @stavares843 @tooshel and I all had these issues 

**Screenshot:**

https://github.com/user-attachments/assets/cda87a8c-f13f-4ed1-aa8a-722623550fd8

image

### PR DESCRIPTION
**Pull request by lgmarchi**
### What this PR does 📖

This pull request introduces several enhancements and bug fixes to the `src/lib/media/Voice.ts` file, including the addition of voice activity detection, improvements to relay server testing, and updates to audio handling. Additionally, there are updates to the `package.json` and `vite.config.js` files to support these changes.

### Voice Activity Detection:
* Added the `voice-activity-detection` library to detect when a user is speaking and update the media state accordingly. [[1]](diffhunk://#diff-a75896c18790a15b7e0ae422d555a80c3e09067296e47a5dcc6891ab3385dfb4L8-R14) [[2]](diffhunk://#diff-a75896c18790a15b7e0ae422d555a80c3e09067296e47a5dcc6891ab3385dfb4R132-L136)

### Relay Server Testing:
* Introduced a method to test the connectivity of relay servers and update the list of available relays dynamically. [[1]](diffhunk://#diff-a75896c18790a15b7e0ae422d555a80c3e09067296e47a5dcc6891ab3385dfb4R30-R43) [[2]](diffhunk://#diff-a75896c18790a15b7e0ae422d555a80c3e09067296e47a5dcc6891ab3385dfb4R639-R697)

### Audio Handling:
* Simplified audio constraints by removing specific audio settings such as echo cancellation and noise suppression.
* Added handling for receiving tracks from peers and logging the events.

### Configuration Updates:
* Updated `package.json` to include the `voice-activity-detection` library.
* Modified `vite.config.js` to include `voice-activity-detection` in optimized dependencies.

### Which issue(s) this PR fixes 🔨

- Resolve #821

### CODE REVIEW

**Comment by phillsatellite** [comment]
Hey Lucas! So during testing Sara, Luis and I have found that friending is not working on this pr. Due to this its not possible to test the talking indicator 

We tried old accounts and new accounts, and made sure to do npm i 

User A sends UserB a friend request 
User B accepts request 
User A appears on User B side but User B does not appear on User A side 

We tried moving to different page, refreshing browser and nothing gets User B to appear on User A side

I also tried 2 new accounts before testing this and 2 new accounts before verifying on dev that friending still worked so clean builds before attempting both 

Your branch:

https://github.com/user-attachments/assets/7b3f4103-0874-42ae-916c-6de7e5ce4400

image

Dev branch:
image

**Review by stavares843 — CHANGES_REQUESTED** [review]
phil added a comment above

**Comment by luisecm** [comment]
> Hey Lucas! So during testing Sara, Luis and I have found that friending is not working on this pr. Due to this its not possible to test the talking indicator
> 
> We tried old accounts and new accounts, and made sure to do npm i
> 
> User A sends UserB a friend request User B accepts request User A appears on User B side but User B does not appear on User A side
> 
> We tried moving to different page, refreshing browser and nothing gets User B to appear on User A side
> 
> I also tried 2 new accounts before testing this and 2 new accounts before verifying on dev that friending still worked so clean builds before attempting both
> 
> Your branch:
> 
>  Screen_Recording_2024-11-08_at_5.50.39_PM.mov 
> 
> image
> 
> Dev branch: image

Hello @lgmarchi just wanted to add that the automated tests are failing against this branch for the same reason mentioned above by Phil. I have not seen this error presented in the latest runs of automation tests against dev or any other prs

**Comment by lgmarchi** [comment]
> Hey Lucas! So during testing Sara, Luis and I have found that friending is not working on this pr. Due to this its not possible to test the talking indicator
> 
> We tried old accounts and new accounts, and made sure to do npm i
> 
> User A sends UserB a friend request User B accepts request User A appears on User B side but User B does not appear on User A side
> 
> We tried moving to different page, refreshing browser and nothing gets User B to appear on User A side
> 
> I also tried 2 new accounts before testing this and 2 new accounts before verifying on dev that friending still worked so clean builds before attempting both
> 
> Your branch:
> 
>  Screen_Recording_2024-11-08_at_5.50.39_PM.mov 
> image
> 
> Dev branch: image

I pushed a commit to solve it.

File: vite.config.js [hunk 1]
```diff
@@ -39,8 +39,10 @@ export default defineConfig({
         nodePolyfills(),
     ],
     optimizeDeps: {
+        include: ["voice-activity-detection"],
         exclude: ["warp-wasm"],
     },
+
```
Comment (stavares843) [inline vite.config.js:45]: do we need the empty line here?

File: src/lib/media/Voice.ts [hunk 2]
```diff
@@ -604,15 +637,64 @@ export class VoiceRTC {
         return accepted
     }
 
+    /**
```
Comment (stavares843) [inline src/lib/media/Voice.ts:641]: the code comments seems a bit extensive

**Comment by phillsatellite** [comment]
So the talking indicator is working a lot better but we ran into some other issues when testing (Attempted with new accounts before testing pr, then verified on dev with the same accounts + new ones)

**Issue:**
It seems like after the first call is made and connected, any call after that don't seem to go through. I tried replicating this issue on dev but the dev branch works fine 

**Steps to reproduce:** 
- UserA calls User B - then after call is connected either person hangs up 
- UserB calls back UserA - call does not go through 
- UserA now tries to call UserB again and call does not go through

**Logs:**
UserA
UserA

UserB
UserB

**Screen recordings:**
This branch 

https://github.com/user-attachments/assets/e5b4a8af-f82d-4f50-bf7d-8b4bdd6444e9

Dev branch

https://github.com/user-attachments/assets/83f12535-9bb4-4a55-a01b-d155034e2ec3

**Comment by luisecm** [comment]
I see that friend requests are working again within latest commit

**Comment by lgmarchi** [comment]
> So the talking indicator is working a lot better but we ran into some other issues when testing (Attempted with new accounts before testing pr, then verified on dev with the same accounts + new ones)
> 
> **Issue:** It seems like after the first call is made and connected, any call after that don't seem to go through. I tried replicating this issue on dev but the dev branch works fine
> 
> **Steps to reproduce:**
> 
> * UserA calls User B - then after call is connected either person hangs up
> * UserB calls back UserA - call does not go through
> * UserA now tries to call UserB again and call does not go through
> 
> **Logs:** UserA UserA
> 
> UserB UserB
> 
> **Screen recordings:** This branch
> 
>  Branch.mov 
> Dev branch
> 
>  DevBranch.mov

I pushed a commit that should solve that.

**Review by stavares843 — COMMENTED** [review]
so like @phillsatellite and i tested and i was talking to phil audio was pretty high but on my user there was zero indicator on my user

image


there was on phil side though and on phil side when he spoke he mentioned he had a talking indicator on his user

**Comment by lgmarchi** [comment]
> so like @phillsatellite and i tested and i was talking to phil audio was pretty high but on my user there was zero indicator on my user
> 
> image
> 
> there was on phil side though and on phil side when he spoke he mentioned he had a talking indicator on his user

So the problem is, when you are talking, no indicator appears to you for your user?

**Comment by stavares843** [comment]
that is correct

**Comment by lgmarchi** [comment]
> that is correct

Which browser? And is it private?

**Comment by stavares843** [comment]
chrome, macOS

private browser yes

**Comment by lgmarchi** [comment]
> chrome, macOS
> 
> private browser yes

Try on normal browser, I am not sure on private it should works nice.

I did 3 tests now with @tooshel , worked all of them, including private browser too.

More thoughts about it? Want to test with other users?

**Review by stavares843 — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Comment by stavares843** [comment]
we approved for now, maybe we can improve in the future

**Review by ashneverdawn — APPROVED** [review]

**Pull request merged**